proxy/proxy.go

```diff
@@ -0,0 +1,705 @@
+package proxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	G "gopkg.in/gilmour-libs/gilmour-e-go.v4"
+	"gopkg.in/gilmour-libs/gilmour-e-go.v4/backends"
+)
+
+// LogError prints error on console, returns string format of the error and raises a panic
+func LogError(err error) string {
+	panic(err)
+	log.Println(err.Error())
+	return err.Error()
+}
+
+// Status is a integer to hold status of node of the node (unavailable = 0, ok = 1 and dirty = 2)
+type Status int
+
+// Status Constant
+const (
+	Unavailable Status = 0 + iota
+	Ok
+	Dirty
+)
+
+var status = [...]string{"unavailable", "ok", "dirty"}
+
+// String returns the English name of the status ("unavailable", "ok", ...).
+func (s Status) String() string { return status[s] }
+
+// NodeID is a string to hold node's id
+type NodeID string
+
+// GilmourTopic is a string to hold topic for signal or request
+type GilmourTopic string
+
+// implements NodeMapOperations
+type nodeMap struct {
+	sync.Mutex
```

Comment:
Shouldn't you use a RWMutex instead of a Mutex? That way Read Locks can be distinguished from a write Lock.